src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -31,22 +33,106 @@ const ActorInit Item_Etcetera_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/ItemEtcetera_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/ItemEtcetera_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/ItemEtcetera_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/func_80920044.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/func_8092009C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/func_809200F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/ItemEtcetera_Update.s")
+static s16 D_80920230[] = {
+    OBJECT_GI_BOTTLE, OBJECT_GI_BOTTLE, OBJECT_GI_BOTTLE,  OBJECT_GI_BOTTLE, OBJECT_GI_BOTTLE,
+    OBJECT_GI_BOTTLE, OBJECT_GI_KEY,    OBJECT_GI_M_ARROW, OBJECT_GI_RUPY,   OBJECT_GI_RUPY,
+    OBJECT_GI_RUPY,   OBJECT_GI_RUPY,   OBJECT_GI_HEARTS,  OBJECT_GI_KEY,
+};
+static s16 D_8092024C[] = {
+    GID_BOTTLE, GID_BOTTLE, GID_BOTTLE, GID_BOTTLE, GID_BOTTLE, GID_BOTTLE, GID_01,
+    GID_47,     GID_4F,     GID_50,     GID_51,     GID_53,     GID_13,     GID_01,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/func_80920164.s")
+static s16 D_80920268[] = {
+    GI_5A, GI_5A, GI_5A, GI_5A, GI_5A, GI_5A, GI_KEY_SMALL, GI_25, GI_NONE, GI_NONE, GI_NONE, GI_NONE, GI_NONE, GI_NONE,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Item_Etcetera/func_809201BC.s")
+void ItemEtcetera_SetupAction(ItemEtcetera* this, ItemEtceteraActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ItemEtcetera_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemEtcetera* this = THIS;
+    s32 pad;
+    s32 params;
+    s32 index;
+
+    params = ITEMETCETERA_GET_FF(&this->actor);
+    index = Object_GetIndex(&globalCtx->objectCtx, D_80920230[params]);
+    if (index >= 0) {
+        this->objIndex = index;
+    }
+    this->itemDrawIndex = D_8092024C[params];
+    this->itemID = D_80920268[params];
+    this->funcSetup = func_8092009C;
+    this->drawFunc = func_809201BC;
+    Actor_SetScale(&this->actor, 0.25f);
+    ItemEtcetera_SetupAction(this, func_80920044);
+    switch (params) {
+        case 7:
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actor.draw = NULL;
+            this->actor.shape.yOffset = 50.0f;
+            break;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+        case 12:
+        case 13:
+            Actor_SetScale(&this->actor, 0.5f);
+            this->funcSetup = func_809200F8;
+            this->drawFunc = func_80920164;
+            this->actor.world.pos.y += 15.0f;
+            break;
+    }
```

Comment:
I don't suppose this can have a default case? Make an enum for the different types too, might as well start it at 0 since the first parts of the tables exist.

Reply:
i didn't get what you tried to say with the "default case", also don't know what to name the enums

Reply:
A default case is when you have
```c
default:
```

in the switch, it handles everything that's not specified. Sometimes not including it can produce strange results (like asschest in OoT, if you've heard of that; it's documented in EnGe1 in OoT). So I wanted to know if this still matched if you added a
```c
default: 
    break;
```

to it.

According to @tom-overton 's research in Discord, the enum should be the same as OoT's for this actor but with the top 6 cases as bottles, compare the `OBJECT_GI_`s in the arrays in the different games https://discord.com/channels/688807550715560050/688851997352263820/987455454521221170